app/Lib/Tools/SyncTool.php

```diff
@@ -60,4 +60,148 @@ public function createHttpSocket($params = array())
         }
         return $HttpSocket;
     }
+
+    /**
+     * @param array $server
+     * @return array|void
+     * @throws Exception
+     */
+    public function getServerRemoteCertificateInfo(array $server)
+    {
+        $httpSocket = $this->setupHttpSocket($server, 5);
```

Comment:
Is the 5sec timeout on purpose or is there a reason not to lower it?
Because by putting the header `Connection: Keep-Alive`, `$httpSocket->get ...` only return after 5sec. So it adds an overhead of 5 sec everytime you do a test even though the connection might return directly.